### PR TITLE
ARROW-15117: [Docs] Splitting the sphinx-based Arrow docs into separate sphinx projects

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Comment by danielfrg** [comment]
The sphinx-tabs package in conda-forge is updated now so you should be able to add it back to the CI :) 

https://github.com/apache/arrow/blob/master/ci/conda_env_sphinx.txt#L25-L27

**Comment by jorisvandenbossche** [comment]
Thanks for that!

**Comment by kszucs** [comment]
Can we build only the relevant sphinx projects in the CI scripts, like in https://github.com/apache/arrow/blob/master/ci/scripts/python_build.sh#L63 ?

**Comment by jorisvandenbossche** [comment]
I put the current on github pages as a preview, see eg the contributing page at https://jorisvandenbossche.github.io/arrow-docs-preview/html-splitted/developers/contributing.html 
(the main index page (https://jorisvandenbossche.github.io/arrow-docs-preview/html-splitted/) should still be updated to get a nicer landing page linking the different docs together)

File: docs/source/conf_common.py [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+
+    subproject = context.get("subproject")
+
+    if not subproject:
+        directory = pagename.split("/")[0]
+
+        subproject = {
+            "format": "Specifications and Protocols",
+            "developers": "Developer documentation",
+            "index": "Home",
```
Comment (jorisvandenbossche) [inline docs/source/conf_common.py:441]: Need to add Java here

**Comment by AlenkaF** [comment]
Ugh, this rebase didn't turn out as good as I thought it would 😬 🤦‍♀️ 😅 
@jorisvandenbossche not sure if I wanna go into resetting this branch or just open a new PR. Sorry for the fuss everybody!

**Comment by AlenkaF** [comment]
Closing in favour of https://github.com/apache/arrow/pull/35610

**Pull request closed without merging**